tree/tree/src/TBranch.cxx

```diff
@@ -1446,6 +1454,66 @@ TFile* TBranch::GetFile(Int_t mode)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Drops the cluster two behind the current cluster and returns a fresh basket
+/// by either reusing or creating a new one
+
+TBasket* TBranch::GetFreshCluster()
+{
+   TBasket *basket = 0;
+
+   // If GetClusterIterator is called with a negative entry then GetStartEntry will be 0
+   // So we need to check if we reach the zero before we have gone back two clusters
+   // if this is the case, we want to keep everything in memory so we return a new basket
+   TTree::TClusterIterator iter = fTree->GetClusterIterator(fBasketEntry[fReadBasket]);
+   if (iter.GetStartEntry() == 0) return fTree->CreateBasket(this);
+
+   for (Int_t j = 0; j < (fTree->GetMaxVirtualSize() * (-1)); j++) {
```

Comment:
This is to allow keeping more than one additional cluster in memory. Brian asked for this. Line 1470 is a little clumsy and should be:
`
   // Iterate backwards (1-VirtualSize) clusters to reach cluster to be unloaded from memory
   for (Int_t j = 0; j < -fTree->GetMaxVirtualSize(); j++) {
`
I'll change that.